src/storage/compactor/src/server.rs

```diff
@@ -84,10 +84,18 @@ pub async fn prepare_start_parameters(
     let non_reserved_memory_bytes = (system_memory_available_bytes() as f64
         * config.storage.compactor_memory_available_proportion)
         as usize;
-    let meta_cache_capacity_bytes = storage_opts.meta_cache_capacity_mb * (1 << 20);
+    let meta_cache_capacity_bytes = storage_opts.compactor_meta_cache_capacity_mb * (1 << 20);
```

Comment:
Given that `extract_storage_memory_config` is only used by compactor, I think it is better to directly refactor and rename `extract_storage_memory_config`.

Reply:
ok, rename `extract_storage_memory_config` to  `extract_storage_memory_config_for_test` and add a new function for `compactor` memory config extract